File: frontend/src/app/shared/service/admin/emailgroup/GroupeEmailDetailAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {GroupeEmailDetailDto} from 'src/app/shared/model/emailgroup/GroupeEmailDetail.model';
import {GroupeEmailDetailCriteria} from 'src/app/shared/criteria/emailgroup/GroupeEmailDetailCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class GroupeEmailDetailAdminService extends AbstractService<GroupeEmailDetailDto, GroupeEmailDetailCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    public constrcutDto(): GroupeEmailDetailDto {
        return new GroupeEmailDetailDto();
    }

    public constrcutCriteria(): GroupeEmailDetailCriteria {
        return new GroupeEmailDetailCriteria();
    }

    get API() {
        return environment.apiUrlEmaillingservice + 'admin/groupeEmailDetail/';
    }
}

File: backend-ms1/backend-ms1/src/main/java/ma/zs/emailling/service/facade/admin/emailgroup/GroupeEmailDetailAdminService.java
package ma.zs.emailling.service.facade.admin.emailgroup;

import java.util.List;
import ma.zs.emailling.bean.core.emailgroup.GroupeEmailDetail;
import ma.zs.emailling.dao.criteria.core.emailgroup.GroupeEmailDetailCriteria;
import ma.zs.emailling.zynerator.service.IService;



public interface GroupeEmailDetailAdminService extends  IService<GroupeEmailDetail,GroupeEmailDetailCriteria>  {

    List<GroupeEmailDetail> findByPersonneId(Long id);
    int deleteByPersonneId(Long id);
    long countByPersonneId(Long id);
    List<GroupeEmailDetail> findByGroupeEmailId(Long id);
    int deleteByGroupeEmailId(Long id);
    long countByGroupeEmailId(Long id);




}
